modules/apps/configuration-admin/configuration-admin-web/src/main/java/com/liferay/configuration/admin/web/internal/servlet/taglib/InstanceSettingsClientExtensionsDynamicInclude.java

```diff
@@ -0,0 +1,244 @@
+
+			List<GlobalJSCET> globalJSCETs = _filterByScriptLocation(
+				_fetchGlobalJSCETs(
+					globalJSClientExtensionsExternalReferenceCodes),
+				scriptLocation);
+
+			for (GlobalJSCET globalJSCET : globalJSCETs) {
+				Map<String, String> scriptAttributes = _toAttributesMap(
+					globalJSCET, httpServletRequest);
+
+				String htmlJSScript = HTMLUtil.toHTMLJSScript(
+					scriptAttributes,
+					_toScriptElementAttributes(
+						globalJSCET.getScriptElementAttributesJSON()));
+
+				printWriter.print(htmlJSScript);
+			}
+		}
+		catch (ConfigurationException configurationException) {
+			_log.error(configurationException);
+
+			throw new RuntimeException(configurationException);
+		}
+	}
+
+	private List<GlobalJSCET> _fetchGlobalJSCETs(
+		String[] globalJSClientExtensionsExternalReferenceCodes) {
+
+		List<GlobalJSCET> globalJSCETS = new ArrayList<>();
+
+		for (String externalRefenceCode :
+				globalJSClientExtensionsExternalReferenceCodes) {
+
+			GlobalJSCET globalJSCET = _getGlobalJSCET(externalRefenceCode);
+
+			if (Objects.nonNull(globalJSCET)) {
+				globalJSCETS.add(globalJSCET);
+			}
+		}
+
+		return globalJSCETS;
+	}
+
+	private List<GlobalJSCET> _filterByScriptLocation(
+		List<GlobalJSCET> globalJSCETS, String scriptLocation) {
+
+		List<GlobalJSCET> filteredGlobalJSCETs = new ArrayList<>();
+
+		for (GlobalJSCET globalJSCET : globalJSCETS) {
+			if (Objects.equals(
+					_getCETScriptLocation(globalJSCET), scriptLocation)) {
+
+				filteredGlobalJSCETs.add(globalJSCET);
+			}
+		}
+
+		return filteredGlobalJSCETs;
+	}
+
+	private String _getCETScriptLocation(GlobalJSCET cet) {
+		return UnicodePropertiesBuilder.fastLoad(
+			cet.getTypeSettings()
+		).build(
+		).get(
+			"scriptLocation"
+		);
+	}
+
+	private GlobalJSCET _getGlobalJSCET(String externalReferenceCode) {
+		return (GlobalJSCET)_cetManager.getCET(
+			CompanyThreadLocal.getCompanyId(), externalReferenceCode);
+	}
+
+	private InstanceSettingsClientExtensionsConfiguration
+			_getInstanceSettingsClientExtensionsConfiguration()
+		throws ConfigurationException {
+
+		return _configurationProvider.getCompanyConfiguration(
+			InstanceSettingsClientExtensionsConfiguration.class,
+			CompanyThreadLocal.getCompanyId());
+	}
+
+	private Map<String, String> _toAttributesMap(
+		GlobalJSCET globalJSCET, HttpServletRequest httpServletRequest) {
+
+		UnicodeProperties typeSettingsUnicodeProperties =
+			UnicodePropertiesBuilder.fastLoad(
+				globalJSCET.getTypeSettings()
+			).build();
+
+		return HashMapBuilder.put(
+			"loadType",
+			() -> {
+				String loadType = typeSettingsUnicodeProperties.getProperty(
+					"loadType");
+
+				if (Objects.equals(loadType, "default")) {
+					return StringPool.BLANK;
+				}
+
+				return loadType;
+			}
+		).put(
+			"nonce",
+			ContentSecurityPolicyNonceProviderUtil.getNonce(httpServletRequest)
+		).put(
+			"src", globalJSCET.getURL()
+		).build();
+	}
+
+	private String _toScriptElementAttributes(
```

Comment:
This looks like it's copied from BaseDynamicInclude in client-extensions-service... Is it possible to share this instead of duplicating it?

Reply:
We haven't found one yet, I'm working on that